iOS/SceneDelegate.swift

```diff
@@ -89,6 +92,96 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 		coordinator.cleanUp(conditional: conditional)
 	}
 	
+	// Handle Opening of URLs
+	
+	func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
+		DispatchQueue.main.async {
+			for context in URLContexts {
+				// Show Unread View or Article
+				if context.url.absoluteString.contains(WidgetDeepLink.unread.url.absoluteString) {
+					guard let comps = URLComponents(string: context.url.absoluteString ) else { return  }
+					let id = comps.queryItems?.first(where: { $0.name == "id" })?.value
+					if id != nil {
+						if AccountManager.shared.isSuspended {
+							AccountManager.shared.resumeAll()
+						}
+						let articles = try! AccountManager.shared.fetchArticles(.unread)
+						let article = articles.filter({ $0.articleID == id }).first
+						
+						if article != nil {
+							self.coordinator.selectAllUnreadFeed()
+							DispatchQueue.main.asyncAfter(deadline: .now() + 0.5, execute: {
+								self.coordinator.selectArticle(article!, animations: [.navigation, .scroll])
+							})
```

Comment:
Instead of doing an `asyncAfter` we can change `selectAllUnreadFeed` and the other smart feed selects to have completion handlers.  The `selectFeed` function they all rely on has a completion handler on it already.  You would just have to pass the new one to it.  You should then be able to write something like this:

```swift
self.coordinator.selectAllUnreadFeed() {
   self.coordinator.selectArticleInCurrentFeed(articleID)
}
```

Reply:
This is what is looks like now:

```swift
if id != nil {
	if AccountManager.shared.isSuspended {
		AccountManager.shared.resumeAll()
	}
	self.coordinator.selectAllUnreadFeed(completion: {
		self.coordinator.selectArticleInCurrentFeed(id!)
	})
} else {
	self.coordinator.selectAllUnreadFeed()
}
```

However, it still needs a dispatch after delay because it's trying to select the article too quickly. What happens at the moment is that it select the unread feed and then just sits there. With the delay it then moves to the article after the initial unread animation has completed.